src/main/java/com/minerarcana/floralchemy/proxy/ClientProxy.java

```diff
@@ -1,16 +1,33 @@
 package com.minerarcana.floralchemy.proxy;
 
 import com.minerarcana.floralchemy.Floralchemy;
-import com.minerarcana.floralchemy.block.flower.SubTilePetroPetunia;
+import com.minerarcana.floralchemy.StateMapperCrystalthorn;
+import com.minerarcana.floralchemy.api.FloralchemyAPI;
+
+import net.minecraft.block.Block;
 import net.minecraft.client.renderer.block.model.ModelResourceLocation;
 import net.minecraft.util.ResourceLocation;
-import vazkii.botania.api.BotaniaAPIClient;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.client.event.ModelRegistryEvent;
+import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.fml.common.Mod.EventBusSubscriber;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
+import net.minecraftforge.fml.relauncher.Side;
 
+@EventBusSubscriber(value = Side.CLIENT, modid = Floralchemy.MOD_ID)
 public class ClientProxy implements IProxy {
-    @Override
-    public void registerModels() {
-        BotaniaAPIClient.registerSubtileModel(SubTilePetroPetunia.NAME,
-                new ModelResourceLocation(new ResourceLocation(Floralchemy.MOD_ID, SubTilePetroPetunia.NAME), "normal"),
-                new ModelResourceLocation(new ResourceLocation(Floralchemy.MOD_ID, SubTilePetroPetunia.NAME), "inventory"));
+	@SubscribeEvent
+    public static void onModelRegistry(ModelRegistryEvent event) {
+        for(Tuple<ResourceLocation, Integer> crystal : FloralchemyAPI.getCrystalRegistry().getCrystals()) {
+        	ResourceLocation crystalthorn = new ResourceLocation(Floralchemy.MOD_ID + ":" + "crystalthorn_" + crystal.getFirst().getPath());
+        	Block block = ForgeRegistries.BLOCKS.getValue(crystalthorn);
+    		if(block == null) {
```

Comment:
Returns in voids are grrrroooosssss. Can easily flip this to remove the return.

Reply:
OK###